### ISSUE TITLE
Custom colored toolbar button icons ignore the disabled setting

### ISSUE THREAD

**Issue by skateman** [patternfly/patternfly-3#1035]
We need to support custom colored button icons in ManageiQ, but it seems like if I have to set the opacity on the color manually if I want to have it look consistently disabled with the button text.

It looks like:
![](https://user-images.githubusercontent.com/649130/39052506-80ca87b6-44ac-11e8-86a5-b73149512e05.png)

But it should look like:
![](https://user-images.githubusercontent.com/649130/39052510-8590cd78-44ac-11e8-8c27-8fa7022ded83.png)

Is there any way to adjust the `disabled` class in patternfly to apply on the custom colored fonticons?

Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1501114
Workaround: https://github.com/ManageIQ/ui-components/pull/284

**Comment by terezanovotna** [comment]
@LHinson any idea who's a good person to ping about this?

**Comment by LHinson** [comment]
@skateman @terezanovotna To confirm, the icons do not look disabled because of this customization? If you used the standard color, would the icon look disabled? 

From a user experience perspective, I don't see any reason why edit, policy or download (as seen in the screenshot above) require a color. What value does that provide? The goal of PatternFly is to provide consistency across the various UIs, therefore I'd recommend removing this customization and opt for the PF recommended colors to align with the broader set of products. 

@terezanovotna @skateman

**Comment by skateman** [comment]
@LHinson the problem here is inconsistency, if the button has no custom color (i.e. has the default one) and it becomes disabled, its color will be adjusted:
![screenshot from 2018-07-08 08-14-53](https://user-images.githubusercontent.com/649130/42417304-2fbb2182-8287-11e8-996c-d32c83f0d3f2.png)

**Comment by terezanovotna** [comment]
@LHinson has a great point about color. It's nice to display color, but it doesn't provide any value to the user, and therefore having the icons black (for active) and grey (for disabled) makes sense. 

@skateman would that resolve the issue?

**Comment by skateman** [comment]
@terezanovotna @LHinson this was a requested feature by @Loicavenel

**Comment by Loicavenel** [comment]
@skateman I cannot remember this, I know this was requested for Custom Button

**Comment by skateman** [comment]
@Loicavenel yes, this is the custom buttons feature with custom colors. As far as I understand @terezanovotna is suggesting to drop the coloring and have all icons gray, so we don't have to deal with the opacity when disabled.

**Comment by terezanovotna** [comment]
@skateman is right. Would it be a reasonable solution? @Loicavenel

**Comment by Loicavenel** [comment]
@terezanovotna we introduced this feature in CF 4.6 and people really like the capability to set color on the 811 icons option, not sure we should removed a feature already supported

**Comment by skateman** [comment]
So the thing I am proposing here is to adjust the definition of the `.diabled` class on a button's icon. Right now PF [explicitly sets a color](https://github.com/patternfly/patternfly/blob/master/src/less/buttons.less#L16) with the evil `!important`. Instead of this the opacity should be set to let's say `50%` on the color that is being set on the non-disabled button. This way it is not important what color is being set for the icon, it will have the disabled look & feel.

**Comment by Loicavenel** [comment]
Seems ok for me..

**Comment by andresgalante** [comment]
Hi @skateman you are right, it should be an opacity. I have 2 concerns:

1.  if we make it a transparency of an unknown color, let's make sure the percentage we pick is readable and accessible.
1. Youy are riught, there shouldn't be an important there and looking in the git history I can't spot how did it end up there. Having said we should make sure that we are not breaking anything by removing it.

@skateman will you be able to send a PR to fix this?

**Comment by skateman** [comment]
@andresgalante I was looking into this, but couldn't come up with a fix, that's why I created the issue and not a PR :disappointed:

**Comment by LHinson** [comment]
@andresgalante we'll need to take this one.
